sdk/python/kfp/components/_data_passing.py

```diff
@@ -107,12 +107,12 @@ def _deserialize_base64_pickle(s):
 
 
 _converters = [
-    Converter([str], ['String', 'str'], _serialize_str, 'str', None),
-    Converter([int], ['Integer', 'int'], _serialize_int, 'int', None),
-    Converter([float], ['Float', 'float'], _serialize_float, 'float', None),
-    Converter([bool], ['Boolean', 'Bool', 'bool'], _serialize_bool, _bool_deserializer_code, _bool_deserializer_definitions),
-    Converter([list], ['JsonArray', 'List', 'list'], _serialize_json, 'json.loads', 'import json'), # ! JSON map keys are always strings. Python converts all keys to strings without warnings
-    Converter([dict], ['JsonObject', 'Dictionary', 'Dict', 'dict'], _serialize_json, 'json.loads', 'import json'), # ! JSON map keys are always strings. Python converts all keys to strings without warnings
+    Converter([str,'str'], ['String', 'str'], _serialize_str, 'str', None),
```

Comment:
Is this change is needed?

Reply:
Yes, this is needed to convert the type 'str' to actual type 'String'. Otherwise, the input as `'str'` will be identified as None type

Reply:
I'm not sure if this is the right place to change though. It looks to me the code build around this won't really make use of the added "type".

For example, `type_name_to_type` will only use the first item in this list:
https://github.com/kubeflow/pipelines/blob/bdb68144496a69ff2e3c30c4986bf88d9ffebc83/sdk/python/kfp/components/_data_passing.py#L122

And for `type_to_type_name`, you'll get both `{str, 'String'}` and `{'str': 'String'}`
https://github.com/kubeflow/pipelines/blob/bdb68144496a69ff2e3c30c4986bf88d9ffebc83/sdk/python/kfp/components/_data_passing.py#L121
But then the key `'str'` (with quotes) will never be used because `type(value)` will never return `'str'`. 
https://github.com/kubeflow/pipelines/blob/bdb68144496a69ff2e3c30c4986bf88d9ffebc83/sdk/python/kfp/components/_data_passing.py#L169

I assume you got an error without changing this? maybe we can sync offline and take a look at the callstack to see if there's a better way to address that.